### ISSUE TITLE
Semantic clarity lacking for assistive technologies, accessibility in design

### ISSUE THREAD

**Issue by AVbiser** [zetkin/app.zetkin.org#2825]
### The feature:
Semantic clarity lacking for assistive technologies which is an accessibility issue

### The bug:
Interactive elements lack sufficient semantic context for assistive technology users since people with disabilities often navigate content without a mouse using the tab key or other key commands to move around the page. When there is repeating information between related pages, it is tedious to tab through these to get to other areas of the page. When repeated content is organized into logical groups that can be easily skipped, navigation becomes much more efficient.

## Steps to reproduce:
1. Go to https://app.dev.zetkin.org/o/1
2. Right click and open "Inspect"
3. Notice all the <div> elements nested in each other and no ARIA lables

### Expected Behaviour:
Suggested solution
-Provide headings for each section using <h1> ... <h6> instead of only using <div> elements
-Implement clear ARIA labels or roles. 
For example:
<button aria-label="Connect and sign up for handing out flyers on June 6, 2025">
    Connect & sign up
</button>

## Screenshots
<img width="748" alt="Image" src="https://github.com/user-attachments/assets/4aacaeea-e9d7-4f12-8675-072ace78c9eb" />
